src/Symfony/Component/Mime/Test/Constraint/EmailHtmlBodyContains.php

```diff
@@ -43,7 +43,7 @@ protected function matches($message): bool
             throw new \LogicException('Unable to test a message HTML body on a RawMessage or Message instance.');
         }
 
-        return false !== mb_strpos($message->getHtmlBody(), $this->expectedText);
+        return false !== mb_strpos((string) $message->getHtmlBody(), $this->expectedText);
```

Comment:
can't happen, only RawMessage is allowed

Reply:
The type hinting of getHtmlBody() says "null|resource|string".

In fact, looking for the type of $message (assuming the phpdoc type is respected - even if it is not strictly enforced with type hinting), we can assume :
- that $message is an "instanceof" RawMessage 
- but not directly a RawMessage or a Message.
Nothing ensures its getText/HtmlBody method returns a string. Nor it even exists.

Maybe this classes would require a refactoring, with creation of an interface ?

Reply:
(but it's outside the scope of this PR)

Reply:
I'm for fixing things that are actual issues.
If you can add a test case for this, I'm fine :)
But them, we should also handle the case where a resource is returned.

Reply:
For the case a resource is returned, the difference is: it won't break _more_ with PHP8.1 or the polyfill, because it will _already_ break.

Again: I checked (at your demand!) what could be checked, i.e when the type error could occur with PHP8.1 although it doesn't occur with previous PHP versions. Even if someone "misuse" the code, like calling addRow with an array of whatever he wants.

All my edits had to reduce inconsistency between current and future php versions / polyfill. I'm just sorry it wouldn't have been understood like that.